test/asn1_time_test.c

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 1999-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Time tests for the asn1 module */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <openssl/asn1.h>
+#include <openssl/evp.h>
+#include <openssl/objects.h>
+#include "test_main.h"
+#include "testutil.h"
+#include "e_os.h"
+
+struct testdata {
+    char *data;  /* TIME string value */
+    int type;    /* GENERALIZED OR UTC */
+    int check;   /* check result */
+    time_t t;    /* expected */
+    int cmp_result;
+    int convert;
+    size_t time_t_size_req;
+};
+
+/* ASSUMES SIGNED TIME_T */
```

Comment:
> According to msdn.

Well, according to Microsoft there [always] is only one version of everything, OS, compiler, language... Latest one. But we don't limit ourselves to latest one alone :-) But 32-bit Windows is not the only problem. Any 32-bit Linux would fail this test, try for example ./Configure linux-x86.

It probably should be clarified that there is chance of misunderstanding about what we are talking about. Kurt might have referred to signness alone, but problem actually goes even beyond 32 bits...